pkg/cli/clienturl/client_url.go

```diff
@@ -274,3 +277,27 @@ func MakeURLParserFn(
 		return clientOpts.ExplicitURL, nil
 	}
 }
+
+// checkMistakes reports likely mistakes to the user via warning messages.
+func (u *urlParser) checkMistakes(purl *pgurl.URL) {
+	if u.warnFn == nil {
+		return
+	}
+
+	// Check mistaken use of -c as direct option, instead of using
+	// options=-c...
+	opts := purl.GetExtraOptions()
+	for optName := range opts {
+		if strings.HasPrefix(optName, "-c") {
+			u.warnFn("\nwarning: found raw URL parameter \"%[1]s\"; "+
+				"are you sure you did not mean to use \"options=%[1]s\" instead?\n\n", optName)
+		}
+	}
+	// For tenant selection, the option is `-ccluster=`, not `-cluster=`.
+	if extraOpts := opts.Get("options"); extraOpts != "" {
+		if strings.Contains(extraOpts, "-cluster=") {
+			u.warnFn("\nwarning: found \"-cluster=\" in URL \"options\" field; " +
```

Comment:
perhaps somewhat confusingly, all of these are allowed:
- `-c cluster=`
- `-ccluster=`
- `--cluster=`

i am not sure if we want to mention the other allowed syntaxes here. wdyt?

Reply:
Can you clarify your question? `-c cluster=` and `-ccluster=` are equivalent (and valid)
I did not know `--cluster=`. Is this one valid?

Reply:
the point i am making is that all three of the syntaxes i listed are valid and equivalent. my question is: should our informational message mention all three of them?

(we matched the PG behavior documented at https://www.postgresql.org/docs/current/app-psql.html - look for `-c command`)

Reply:
a separate point i just thought of: if we have to pick only one to recommend, then i think `options=-ccluster=` is less clear than `options=-c cluster=`

Reply:
i've ended up reusing the same code as pgwire so we have consistent behavior.